File: max-min/max-min.py
#hackerRank - Max min

def maxMin(k, arr):
    # Write your code here
    if k == len(arr):
        return max(arr) - min(arr)
    arr.sort()
    result = 1000000000000
    start = 0
    for end in range(k - 1, len(arr)):
        current_unfair = arr[end] - arr[start]
        if current_unfair < result:
            result = current_unfair
        start += 1
    return result

File: max-min/max-min.js
//hackerRank - Max min

function maxMin(k, arr) {
    // Write your code here
    if (k === arr.length) {
        return Math.max(...arr) - Math.min(...arr)
    };
    const sortedArr = arr.sort((a, b) => a - b);
    let minDiff = 10000000000;
    let start = 0;
    for (let end = k - 1; end < arr.length; end++) {
        let currentDiff = sortedArr[end] - sortedArr[start];
        if (currentDiff < minDiff) {
            minDiff = currentDiff
        };
        start += 1
    };
    return minDiff
}